core/src/main/java/org/radargun/utils/Utils.java

```diff
@@ -249,15 +249,24 @@ public static Properties getPluginProperties(String plugin) {
          log.warn("Could not find a plugin descriptor : " + file);
          return properties;
       }
-      FileInputStream inStream = null;
       try {
-         inStream = new FileInputStream(file);
-         properties.load(inStream);
-         return properties;
+         loadProperties(properties, file);
       } catch (IOException e) {
-         throw new RuntimeException(e);
+         log.error(String.format("Exception while loading plugin specific properties file %s.", PROPERTIES_FILE), e);
+      }
+      return properties;
+   }
+
+   public static Properties loadProperties(Properties properties, File file) throws IOException {
+      return loadProperties(properties, new FileInputStream(file));
+   }
+
+   public static Properties loadProperties(Properties properties, InputStream inputStream) throws IOException {
```

Comment:
I think that closing resource in different scope than where it is allocated is an anti-pattern. RG is baselined on Java 7, use try-with-resources instead, please.